servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/TaskBasedAsyncCompletableOperator.java

```diff
@@ -44,19 +43,19 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(TaskBasedAsyncCompletableOperator.class);
 
     private final Completable original;
-    private final Supplier<? extends BooleanSupplier> shouldOffloadSupplier;
+    private final BooleanSupplier shouldOffload;
     private final Executor executor;
 
     TaskBasedAsyncCompletableOperator(final Completable original,
-                                      final Supplier<? extends BooleanSupplier> shouldOffloadSupplier,
+                                      final BooleanSupplier shouldOffload,
                                       final Executor executor) {
         this.original = original;
-        this.shouldOffloadSupplier = shouldOffloadSupplier;
+        this.shouldOffload = shouldOffload;
         this.executor = executor;
```

Comment:
Comment for all async sources: because `BooleanSupplier` and `Executor` are provided by the user, consider adding `requireNonNull` for these two fields.